controllers/agents/app/serviceclaim_controller.go

```diff
@@ -43,11 +61,141 @@ type ServiceClaimReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	l.Info("Reconciling service claim")
+
+	var sclaim primazaiov1alpha1.ServiceClaim
+	if err := r.Get(ctx, req.NamespacedName, &sclaim); err != nil {
+		l.Info("unable to retrieve ServiceClaim", "error", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	config, remote_namespace, err := r.GetPrimazaKubeconfig(ctx, sclaim.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	l.Info("remote cluster", "address", config.Host)
+
+	remote_client, err := client.New(config, client.Options{
+		Scheme: r.Client.Scheme(),
+		Mapper: r.Mapper,
+	})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	objKey := client.ObjectKey{
+		Name:      constants.ApplicationAgentDeploymentName,
+		Namespace: req.NamespacedName.Namespace,
+	}
+	var deployment appsv1.Deployment
+	if err := r.Get(ctx, objKey, &deployment); err != nil {
+		l.Error(err, "unable to retrieve agent app deployment")
```

Comment:
If we are unable to find the agent app deployment, then we should have panic here instead of returning error to reconciler

Reply:
We may also consider using kubernetes downward api here: https://kubernetes.io/docs/concepts/workloads/pods/downward-api/

Reply:
@Kartikey-star If the deployment is not available at the moment, returning the error will try to reconcile again. I think that's the correct behavior.

Reply:
Done